benchmarks/DeepLearning/Ops/MatMulOp/CMakeLists.txt

```diff
@@ -11,39 +24,42 @@ add_custom_command(OUTPUT matmul-scalar.o
             -llvm-request-c-wrappers
             -convert-func-to-llvm
             -reconcile-unrealized-casts | 
-          ${LLVM_MLIR_BINARY_DIR}/mlir-translate --mlir-to-llvmir |
-          ${LLVM_MLIR_BINARY_DIR}/llc -O3 -mtriple=${BUDDY_OPT_TRIPLE} 
-            -mattr=${BUDDY_OPT_ATTR} --filetype=obj 
-	          -o ${BUDDY_BINARY_DIR}/../benchmarks/DeepLearning/Ops/MatMulOp/matmul-scalar.o
+            ${LLVM_MLIR_BINARY_DIR}/mlir-translate --mlir-to-llvmir -o matmul_scalar_O0.ll
+  COMMAND ${LLVM_MLIR_BINARY_DIR}/clang -O0 ${CLANG_FLAGS_LIST} matmul_scalar_O0.ll
+          -c -save-temps -o ${CMAKE_CURRENT_BINARY_DIR}/matmul_scalar_O0.o
 )
-add_library(MatMulScalar STATIC matmul-scalar.o)
-set_target_properties(MatMulScalar PROPERTIES LINKER_LANGUAGE CXX)
-
-add_custom_command(OUTPUT matmul-transform.o
-  COMMAND cat ${BUDDY_SOURCE_DIR}/benchmarks/DeepLearning/Ops/MatMulOp/MatMulTransform.mlir |
-          ${LLVM_MLIR_BINARY_DIR}/mlir-opt
-            -llvm-request-c-wrappers
-            -test-lower-to-llvm |
-          ${LLVM_MLIR_BINARY_DIR}/mlir-translate --mlir-to-llvmir |
-          ${LLVM_MLIR_BINARY_DIR}/llc -O3 -mtriple=${BUDDY_OPT_TRIPLE}
-            -mattr=${BUDDY_OPT_ATTR} --filetype=obj
-	          -o ${BUDDY_BINARY_DIR}/../benchmarks/DeepLearning/Ops/MatMulOp/matmul-transform.o
-)
-add_library(MatMulTransform STATIC matmul-transform.o)
-set_target_properties(MatMulTransform PROPERTIES LINKER_LANGUAGE CXX)
-
-add_executable(dl-op-linalg-matmul-benchmark
-  GoogleBenchmarkMain.cpp
-  )
-
-set_target_properties(dl-op-linalg-matmul-benchmark PROPERTIES
-  LINK_FLAGS "-static"
+add_library(matmul_scalar_O0 STATIC matmul_scalar_O0.o)
+set_target_properties(matmul_scalar_O0 PROPERTIES LINKER_LANGUAGE CXX)
+target_link_libraries(dl-op-linalg-matmul-benchmark
+  matmul_scalar_O0
 )
 
-set(BenchmarkTool GoogleBenchmark)
-
-target_link_libraries(dl-op-linalg-matmul-benchmark
-  ${BenchmarkTool}
-  MatMulScalar
-  MatMulTransform
+macro(add_scalar_lowering_target TARGET_NAME FILE_NAME)
```

Comment:
When we have the lowering pass, we should remove this macro.

Reply:
This macro function is retained because it involves scalar lowering passes, which are needed for comparing our optimization methods. In the future, each subsequent optimization method will add its own lowering pass.